server/base/src/main/java/org/apache/accumulo/server/fs/VolumeManagerImpl.java

```diff
@@ -384,20 +411,61 @@ public short getDefaultReplication(Path path) {
    * @param filesystemURI Volume Filesystem URI
    * @return Hadoop Configuration with custom overrides for this FileSystem
    */
-  private static Configuration getVolumeManagerConfiguration(AccumuloConfiguration conf,
+  protected static Configuration getVolumeManagerConfiguration(AccumuloConfiguration conf,
       final Configuration hadoopConf, final String filesystemURI) {
 
-    final Configuration volumeConfig = new Configuration(hadoopConf);
-
-    conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUME_CONFIG_PREFIX).entrySet()
-        .stream().filter(e -> e.getKey().startsWith(filesystemURI + ".")).forEach(e -> {
-          String key = e.getKey().substring(filesystemURI.length() + 1);
-          String value = e.getValue();
-          log.info("Overriding property {} for volume {}", key, value, filesystemURI);
-          volumeConfig.set(key, value);
-        });
+    Map<String,String> volumeHdfsConfigOverrides =
+        conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUME_CONFIG_PREFIX).entrySet()
+            .stream().filter(e -> e.getKey().startsWith(filesystemURI + ".")).collect(
+                Collectors.toUnmodifiableMap(e -> e.getKey().substring(filesystemURI.length() + 1),
+                    Entry::getValue));
 
-    return volumeConfig;
+    if (volumeHdfsConfigOverrides.isEmpty()) {
+      // There are no overrides in the AccumuloConfiguration for this volume, return the Hadoop
+      // Configuration input parameter.
+      return hadoopConf;
+    } else {
+      while (true) {
+
+        // Get the ConfigurationOverride object from the Cache. If no object exists, then create
+        // one - it's comprised of the volume overrides map from above and a Hadoop Configuration
+        // object with the overrides applied.
+        final AtomicBoolean createdNew = new AtomicBoolean(false);
+        ConfigurationOverride cachedVolumeConfig =
+            HDFS_CONFIGS_FOR_VOLUME.get(filesystemURI, (fs) -> {
+              createdNew.set(true);
+              log.debug("Caching a new configuration for volume: {} with overrides: {}",
+                  filesystemURI, volumeHdfsConfigOverrides);
+              Configuration volumeConfig = new Configuration(hadoopConf);
```

Comment:
This caching something derived from a method parameter, the hadoopConfig. Need to account for different configs somehow or refactor the code to not pass in a hadoopConfig.